File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
            # create stack
            l = []
            for i in s :
                if i in ['(', '[', '{']:
                    l.append(i)
                else:
                    if len(l) == 0:
                        return False
                    else:
                        if (i == ')' and l[-1] == '(' or
                           i == ']' and l[-1] == '[' or
                           i == '}' and l[-1] == '{'):
                            l.pop()
                        else:
                            return False
            if len(l) == 0:
                return True
            else:
                return False

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
     
        
	stack<char> stc;
	for(auto i : s){
		if (i == '(' || i == '{' || i == '['){
			stc.push(i);
		}else{
			if(stc.empty()){
				{
				return false;
				}
			}else{

				if(i == ')' && stc.top() == '('){
					stc.pop();
				}
				else if(i == ']' && stc.top() == '[')
					stc.pop();
				else if(i == '}' && stc.top() == '{')
					stc.pop();
				else
					return false;
			}
		}
	}

	if (stc.empty())
		return true;
	return false;

    }
};